### ISSUE TITLE
ICMP does not work on Mac OS

### ISSUE THREAD

**Issue by xoraingroup** [TwinProduction/gatus#132]
MAC OS Big Sur version 11.0.1

Everything works perfectly ok except icmp in MAC OS.

Config:

  - name: node-1
    url: "icmp://104.237.x.x"
    group: Uptime (Ping)
    conditions: 
      - "[CONNECTED] == true"
      - "[RESPONSE_TIME] > 0"

![image](https://user-images.githubusercontent.com/17129052/125159622-34ee5f00-e181-11eb-90f9-d3d53ac287ed.png)

the x.x is in ip is set intentionally. 

Is this wrong config?

**Comment by TwinProduction** [comment]
I assume that the `x.x` is there to hide your IP -- can you try the following instead:
```yaml
  - name: node-1
    group: Uptime (Ping)
    url: "icmp://127.0.0.1"
    conditions:
      - "[CONNECTED] == true"
      - "[RESPONSE_TIME] < 150"
```

**Comment by xoraingroup** [comment]
Yes correct. 

Tried and following are the results

![image](https://user-images.githubusercontent.com/17129052/125171186-7a2f8280-e1bb-11eb-8c26-09d59dd8e899.png)

![image](https://user-images.githubusercontent.com/17129052/125171224-a8ad5d80-e1bb-11eb-9cae-728f342b051e.png)

Its not working even for localhost.

**Comment by TwinProduction** [comment]
When you say "on Mac OS", do you mean that you're running Gatus as an executable instead of going through Docker?

**Comment by xoraingroup** [comment]
Yes thats correct. I am not using Docker.

**Comment by TwinProduction** [comment]
I don't have a Mac I can test this on, but this is likely where the culprit is:
https://github.com/TwinProduction/gatus/blob/9b68582622e5d20be48652ccc74440b279ad335e/client/client.go#L108-L111

Can you try running Gatus with elevated privileges, and see if the results are the same?
If that doesn't change anything, could you try to rebuild the executable after updating the boolean passed as parameter on L110  to `false`?

**Comment by xoraingroup** [comment]
Yes you are right. With elevated privileges it works. 

![image](https://user-images.githubusercontent.com/17129052/125351082-4e053480-e368-11eb-974c-3318d6a7df9e.png)

Ping should not need any elevated privileges. I think it might need a fix. 

But its also good to know and you can at least mention in repo documentation for other people who might be facing such issue. The rest its awesome tool and thank you for this.

**Comment by xoraingroup** [comment]
Also just rebuilt it with your instruction by setting ping.SetPrivileged(false) and its working fine without sudo. Why did you put it in the first place? Just curious to know.

**Comment by TwinProduction** [comment]
I believe it doesn't work on Windows (or was it Docker?) without `pinger.SetPrivileged(true)`, but I'd have to double check.


Adding a conditional statement checking the value of GOOS should be able to resolve this issue.

**Comment by xoraingroup** [comment]
Ok good to know. Thank you for the fix.

**Comment by TwinProduction** [comment]
@xoraingroup Can you give it a try and let me know if the latest version works?

**Comment by xoraingroup** [comment]
Working perfectly fine. Thank you!

**Comment by TwinProduction** [comment]
Released in [v2.8.0](https://github.com/TwinProduction/gatus/releases/tag/v2.8.0)
